packages/service-clients/azure-client/package.json

```diff
@@ -128,6 +124,7 @@
 		"cross-env": "^7.0.3",
 		"eslint": "~8.55.0",
 		"eslint-config-prettier": "~9.0.0",
+		"fluid-framework": "workspace:~",
```

Comment:
I think this is probably fine since it's a dev dependency and only for use in testing (to make it more authentic to what a customer would do)?

I do think we should continue to avoid taking non-dev dependencies on fluid-framework in our own packages to keep our layering simple.

Reply:
> I do think we should continue to avoid taking non-dev dependencies on fluid-framework in our own packages to keep our layering simple.

I agree, but hereby present what seems like a legit exception to the rule 😄 : https://github.com/microsoft/FluidFramework/pull/23647. That's an e2e tests package though, which arguably should be a `@fluid-internal` one (I'd say `@fluid-private` but I _suspect_ we do need it published internally due to how we run those tests)

Reply:
Oh agree, was mostly thinking about product (shipping) packages.  Anywhere where we are acting "as a customer would" (e2e tests, examples, etc.) the dependency seems fine to me.

Reply:
Agree with all of the above. But I would also argue that any example / end-to-end test packages that use the declarative API should _strictly_ use `fluid-framework`, to help validate that our end-users actually have everything they need to succeed using those APIs.